File: BackEnd/src/domain/dtos/user/update-profile-request.dto.ts
/**
 * @file update-profile-request.dto.ts
 * @description DTO de actualización de perfil de usuario. Se completará en Fase 2.
 * @module User
 * @layer Domain
 */
import { IsString, IsOptional, IsNumber, MaxLength } from 'class-validator';

export class UpdateProfileRequestDto {
  @IsOptional() @IsString() @MaxLength(50)  name?: string;
  @IsOptional() @IsString() @MaxLength(200) bio?: string;
  @IsOptional() @IsString() location?: string;
  /** Latitud de la ubicación seleccionada del catálogo de capitales */
  @IsOptional() @IsNumber() locationLat?: number;
  /** Longitud de la ubicación seleccionada del catálogo de capitales */
  @IsOptional() @IsNumber() locationLon?: number;
  /** URL Cloudinary de la foto de perfil */
  @IsOptional() @IsString() photo?: string;
  /** URL Cloudinary del banner/fondo de perfil */
  @IsOptional() @IsString() bannerPhoto?: string;
}

File: FrontEnd/lib/domain/dtos/user/update_profile_request_dto.dart
/// @file update_profile_request_dto.dart
/// @description DTO para actualizar el perfil del usuario.
/// Todos los campos son opcionales — solo se envían los que cambian.
/// @module User
/// @layer Domain
library;

// ═══════════════════════════════════════════════════════════════════════════════
// UPDATE PROFILE REQUEST DTO
// ═══════════════════════════════════════════════════════════════════════════════

/// DTO de actualización de perfil del usuario.
class UpdateProfileRequestDto {
  const UpdateProfileRequestDto({
    this.name,
    this.bio,
    this.location,
    this.locationLat,
    this.locationLon,
    this.photo,
    this.bannerPhoto,
  });

  final String? name;
  final String? bio;
  final String? location;

  /// Latitud de la capital de provincia seleccionada.
  final double? locationLat;

  /// Longitud de la capital de provincia seleccionada.
  final double? locationLon;

  /// URL Cloudinary de la nueva foto de perfil, o null si no cambia.
  final String? photo;

  /// URL Cloudinary del banner/fondo de perfil, o null si no cambia.
  final String? bannerPhoto;

  Map<String, dynamic> toJson() => {
    if (name        != null) 'name':        name,
    if (bio         != null) 'bio':         bio,
    if (location    != null) 'location':    location,
    if (locationLat != null) 'locationLat': locationLat,
    if (locationLon != null) 'locationLon': locationLon,
    if (photo       != null) 'photo':       photo,
    if (bannerPhoto != null) 'bannerPhoto': bannerPhoto,
  };
}
